vent/controller/control_module.py

```diff
@@ -238,14 +232,18 @@ def update(self, pressure):
             self.cycle_waveforms[self.cycle_counter] = data
 
 
-class ControlModuleSimulator(ControlModuleBase):
+class ControlModuleSimulator(ControlModuleBase, threading.Thread):
     # Implement ControlModuleBase functions
-    def __init__(self):
-        super().__init__()  # get all from parent
+    def __init__(self, threadID, name):
+
+        ControlModuleBase.__init__(self)
+        threading.Thread.__init__(self)
+        self.threadID        = threadID
+        self.name            = name
 
         self.Balloon = Balloon_Simulator(leak=True, delay=False)
         self.Controller = StateController()
-        self.loop_counter = 0
+        self._running = True
```

Comment:
Can you make `_running` default to be False? I feel the use case would click the start button on the UI to start the behavior. And coordinator can also call the `start` function

Reply:
yep, sure

Reply:
done